test/integration/api/allocation/idle_cost_negative_test.go

```diff
@@ -0,0 +1,184 @@
+	}
+	if m.GPUCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "GPUCost")
+	}
+	if m.RAMCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "RAMCost")
+	}
+	if m.NetworkCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "NetworkCost")
+	}
+	if m.LoadBalancerCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "LoadBalancerCost")
+	}
+
+	// Check Idle-Cost fields
+	if m.CPUCostIdle < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "CPUCostIdle")
+	}
+	if m.GPUCostIdle < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "GPUCostIdle")
+	}
+	if m.RAMCostIdle < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "RAMCostIdle")
+	}
+
+	// Check adjustment cost fields
+	if m.CPUCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "CPUCostAdjustment")
+	}
+	if m.GPUCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "GPUCostAdjustment")
+	}
+	if m.NetworkCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "NetworkCostAdjustment")
+	}
+	if m.LoadBalancerCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "LoadBalancerCostAdjustment")
+	}
+
+	return isNegative, negativeFields
+}
+
+func TestNegativeIdleCosts(t *testing.T) {
+	apiObj := api.NewAPI()
+
+	testCases := []struct {
+		name		string
+		window      string
+		aggregate   string
+		accumulate  string
+		includeidle string
+	}{
+		{
+			name: "Today",
+			window: "today",
+			aggregate: "namespace",
+			accumulate: "false",
+			includeidle: "true",
+		},
+		// { // This test is meant to fail because there is no includeidle field, i.e no __idle__
+		// 	name: "Missing includeIdle",
+		// 	window: "today",
+		// 	aggregate: "namespace",
+		// 	accumulate: "false",
+		// },
+		{
+			name: "Yesterday",
+			window: "yesterday",
+			aggregate: "node",
+			accumulate: "false",
+			includeidle: "true",
+		},
+		// {
+		// 	name: "Last week",
+		// 	window: "week",
+		// 	aggregate: "service",
+		// 	accumulate: "false",
+		// 	includeidle: "true",
+		// },
+		// {
+		// 	name: "Last 14 days",
+		// 	window: "14d",
+		// 	aggregate: "pod",
+		// 	accumulate: "false",
+		// 	includeidle: "true",
+		// },
+		// {
+		// 	name: "Custom",
+		// 	window: "%sT00:00:00Z,%sT00:00:00Z", // This can be generated dynamically based on the running time
+		// 	aggregate: "namespace",
+		// 	accumulate: "false",
+		// 	includeidle: "true",
+		// },
+	}
+
+	t.Logf("testCases: %v", testCases)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			// if tc.name == "Custom" {
+			// 	// Dynamically generate the "Custom" window
+			// 	now := time.Now()
+			// 	tc.window = fmt.Sprintf(tc.window,
+			// 		now.AddDate(0, 0, -2).Format("2006-01-02"),
+			// 		now.AddDate(0, 0, -1).Format("2006-01-02"),
+			// 	)
+			// }

```

Comment:
what's this?

Reply:
Code that I needed to make a specific test case work. If I uncomment my other test cases, I can uncomment it